src/Ocelot/DependencyInjection/BuilderExtensions/CoreExtensions.cs

```diff
@@ -0,0 +1,235 @@
+            services.TryAddSingleton<IBaseUrlFinder, BaseUrlFinder>();
+            services.TryAddSingleton<IRegionCreator, RegionCreator>();
+            services.TryAddSingleton<IFileConfigurationRepository, FileConfigurationRepository>();
+            services.TryAddSingleton<IFileConfigurationSetter, FileConfigurationSetter>();
+            services.TryAddSingleton<IFileConfigurationProvider, FileConfigurationProvider>();
+
+            services.TryAddSingleton<IUrlBuilder, UrlBuilder>();
+            services.TryAddSingleton<IRemoveOutputHeaders, RemoveOutputHeaders>();
+            services.TryAddSingleton<IOcelotConfigurationProvider, OcelotConfigurationProvider>();
+            services.TryAddSingleton<IClaimToThingConfigurationParser, ClaimToThingConfigurationParser>();
+            services.TryAddSingleton<IClaimsAuthoriser, ClaimsAuthoriser>();
+            services.TryAddSingleton<IScopesAuthoriser, ScopesAuthoriser>();
+            services.TryAddSingleton<IAddClaimsToRequest, AddClaimsToRequest>();
+            services.TryAddSingleton<IAddHeadersToRequest, AddHeadersToRequest>();
+            services.TryAddSingleton<IAddQueriesToRequest, AddQueriesToRequest>();
+            services.TryAddSingleton<IClaimsParser, ClaimsParser>();
+            services.TryAddSingleton<IUrlPathToUrlTemplateMatcher, RegExUrlMatcher>();
+            services.TryAddSingleton<IUrlPathPlaceholderNameAndValueFinder, UrlPathPlaceholderNameAndValueFinder>();
+            services.TryAddSingleton<IDownstreamPathPlaceholderReplacer, DownstreamTemplatePathPlaceholderReplacer>();
+            services.TryAddSingleton<IDownstreamRouteFinder, DownstreamRouteFinder.Finder.DownstreamRouteFinder>();
+            services.TryAddSingleton<IHttpRequester, HttpClientHttpRequester>();
+            services.TryAddSingleton<IHttpResponder, HttpContextResponder>();
+            services.TryAddSingleton<IRequestCreator, HttpRequestCreator>();
+            services.TryAddSingleton<IErrorsToHttpStatusCodeMapper, ErrorsToHttpStatusCodeMapper>();
+
+            services.TryAddSingleton<IHttpClientCache, MemoryHttpClientCache>();
+            services.TryAddSingleton<IRequestMapper, RequestMapper>();
+            services.TryAddSingleton<IHttpHandlerOptionsCreator, HttpHandlerOptionsCreator>();
+            // see this for why we register this as singleton http://stackoverflow.com/questions/37371264/invalidoperationexception-unable-to-resolve-service-for-type-microsoft-aspnetc
+            // could maybe use a scoped data repository
+            services.TryAddSingleton<IHttpContextAccessor, HttpContextAccessor>();
+            services.TryAddScoped<IRequestScopedDataRepository, HttpDataRepository>();
+            services.AddMemoryCache();
+            services.TryAddSingleton<OcelotDiagnosticListener>();
+
+            return builder;
+        }
+
+        public static IOcelotBuilder AddAspNetCoreServices(this IOcelotBuilder builder) {
```

Comment:
This should be private or internal because users should not be able to call it? Ocelot won't work without this being called?